launcher/minecraft/auth/MinecraftAccount.cpp

```diff
@@ -334,3 +335,32 @@ void MinecraftAccount::incrementUses()
         qWarning() << "Profile" << data.profileId() << "is now in use.";
     }
 }
+
+QUuid MinecraftAccount::uuidFromUsername(QString username) {
+    auto input = QString("OfflinePlayer:%1").arg(username).toUtf8();
+
+    // basically a reimplementation of Java's UUID#nameUUIDFromBytes
+    QByteArray digest = QCryptographicHash::hash(input, QCryptographicHash::Md5);
+
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    auto bOr = [](QByteArray& array, int index, char value) {
+        array[index] = array.at(index) | value;
+    };
+    auto bAnd = [](QByteArray& array, int index, char value) {
+        array[index] = array.at(index) & value;
+    };
+#else
+    auto bOr = [](QByteArray& array, qsizetype index, char value) {
+        array[index] |= value;
+    };
+    auto bAnd = [](QByteArray& array, qsizetype index, char value) {
+        array[index] &= value;
+    };
+#endif
+    bAnd(digest, 6, (char) 0x0f); // clear version
+    bOr(digest, 6, (char) 0x30); // set to version 3
+    bAnd(digest, 8, (char) 0x3f); // clear variant
+    bOr(digest, 8, (char) 0x80); // set to IETF variant
```

Comment:
```suggestion
    digest[6] &= (char) 0x0f; // clear version
    digest[6] |= (char) 0x30; // set to version 3
    digest[8] &= (char) 0x3f; // clear variant
    digest[8] |= (char) 0x80; // set to IETF variant
```

Why doesn't this work? it's more readable

Reply:
Doesn't work because Qt 5 returns a QCharRef that doesn't support the `|` and `&` operators.

Reply:
oops